test/Dialect/FIRRTL/lower-layers.mlir

```diff
@@ -350,6 +350,75 @@ firrtl.circuit "Test" {
     %foo_o = firrtl.instance foo @Foo(out o : !firrtl.probe<uint<1>, @A>)
     %x = firrtl.ref.resolve %foo_o : !firrtl.probe<uint<1>, @A>
   }
+
+  //===--------------------------------------------------------------------===//
+  // Inline Layers
+  //===--------------------------------------------------------------------===//
+
+  // CHECK:  sv.macro.decl @Inline_0["Inline"]
+  firrtl.layer @Inline inline {
+    // CHECK:  sv.macro.decl @Inline_Inline
+    firrtl.layer @Inline inline {}
+    firrtl.layer @Bound bind {
+    // CHECK:  sv.macro.decl @Inline_Bound_Inline
+      firrtl.layer @Inline inline {}
+      firrtl.layer @Bound bind {
+      // CHECK:  sv.macro.decl @Inline_Bound_Bound_Inline
+        firrtl.layer @Inline inline {}
+      }
+    }
+  }
+
+  firrtl.layer @Bound bind {
+    // CHECK: sv.macro.decl @Bound_Inline
+    firrtl.layer @Inline inline {}
+  }
+
+  // CHECK: firrtl.module private @ModuleWithInlineLayerBlocks_Inline_Bound() {
+  // CHECK:   %w3 = firrtl.wire : !firrtl.uint<3>
+  // CHECK:   sv.ifdef  @Inline_Bound_Inline {
+  // CHECK:     %w4 = firrtl.wire : !firrtl.uint<4>
+  // CHECK:   }
+  // CHECK: }
+
+  // CHECK: firrtl.module private @ModuleWithInlineLayerBlocks_Bound() {
+  // CHECK:   %w5 = firrtl.wire : !firrtl.uint<5>
+  // CHECK:   sv.ifdef @Bound_Inline {
+  // CHECK:     %w6 = firrtl.wire : !firrtl.uint<6>
+  // CHECK:   }
+  // CHECK: }
+
+  // CHECK: firrtl.module @ModuleWithInlineLayerBlocks() {
+  // CHECK:   sv.ifdef  @Inline_0 {
+  // CHECK:     %w1 = firrtl.wire : !firrtl.uint<1>
+  // CHECK:     sv.ifdef  @Inline_Inline {
+  // CHECK:       %w2 = firrtl.wire : !firrtl.uint<2>
+  // CHECK:     }
+  // CHECK:     firrtl.instance inline_bound {lowerToBind, output_file = #hw.output_file<"layers_Test_Inline_Bound.sv", excludeFromFileList>} @ModuleWithInlineLayerBlocks_Inline_Bound()
```

Comment:
The ABI of the FIRRTL spec should be updated to match this file name. Note: I think that including the inline layer in the name is fine. The alternative of not including it seems less fine. Regardless, this should be rationalized.